src/workerd/io/actor-sqlite.c++

```diff
@@ -721,6 +721,25 @@ void ActorSqlite::TxnCommitRegulator::onError(
   // as we don't know for sure that the problem is the application's fault.
 }
 
+void ActorSqlite::TxnCommitRegulator::onCriticalError(
```

Comment:
This regulator is only used when committing a transaction. But the error could happen on any statement within the transaction.

I think this shouldn't be a regulator callback. Instead we need something like `SqliteDatabase::onCriticalError()` which lets you specify a callback to call in this case, similar to how `onWrite()` works.